bodhi/server/models.py

```diff
@@ -62,6 +46,167 @@
+                setattr(cls, k, sym)
+        return type.__init__(cls, classname, bases, dict_)
+
+    def __iter__(cls):
+        return iter(cls._reg.values())
+
+
+class DeclEnum(object):
+    """Declarative enumeration."""
+
+    __metaclass__ = EnumMeta
+    _reg = {}
+
+    @classmethod
+    def from_string(cls, value):
+        try:
+            return cls._reg[value]
+        except KeyError:
+            raise ValueError("Invalid value for %r: %r" % (cls.__name__, value))
+
+    @classmethod
+    def values(cls):
+        return cls._reg.keys()
+
+    @classmethod
+    def db_type(cls):
+        return DeclEnumType(cls)
+
+
+class DeclEnumType(SchemaType, TypeDecorator):
+    def __init__(self, enum):
+        self.enum = enum
+        self.impl = Enum(
+            *enum.values(),
+            name="ck%s" % re.sub('([A-Z])', lambda m: "_" + m.group(1).lower(), enum.__name__))
+
+    def _set_table(self, table, column):
+        self.impl._set_table(table, column)
+
+    def copy(self):
+        return DeclEnumType(self.enum)
+
+    def process_bind_param(self, value, dialect):
+        """
+        :type value:   bodhi.server.models.enum.EnumSymbol
+        :type dialect: sqlalchemy.engine.default.DefaultDialect
+        """
+        if value is None:
+            return None
+        return value.value
+
+    def process_result_value(self, value, dialect):
+        if value is None:
+            return None
+        return self.enum.from_string(value.strip())
+
+    def create(self, bind=None, checkfirst=False):
+        """Issue CREATE ddl for this type, if applicable."""
+        super(DeclEnumType, self).create(bind, checkfirst)
+        t = self.dialect_impl(bind.dialect)
+        if t.impl.__class__ is not self.__class__ and isinstance(t, SchemaType):
+            t.impl.create(bind=bind, checkfirst=checkfirst)
+
+    def drop(self, bind=None, checkfirst=False):
+        """Issue DROP ddl for this type, if applicable."""
+        super(DeclEnumType, self).drop(bind, checkfirst)
+        t = self.dialect_impl(bind.dialect)
+        if t.impl.__class__ is not self.__class__ and isinstance(t, SchemaType):
+t.impl.drop(bind=bind, checkfirst=checkfirst)
+
+# encoding: utf-8
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
```

Comment:
Move the docstring before import statement https://www.python.org/dev/peps/pep-0008/#imports
Here is  duplication of docstring https://github.com/ankit01ojha/bodhi/blob/b7fdff2e6f66d8a487848f8424189bec798a55c9/bodhi/server/models.py#L49-L77